modules/ROOT/pages/dataweave-cookbook-extract-data.adoc

```diff
@@ -4,287 +4,200 @@ include::_attributes.adoc[]
-```
-+
-Only returns the first `user` value, `"a"`  if the `output` directive is set
-to `application/json`.
-
-For more examples, see <<single_value>>.
-
-| *Multiple Values*  (`*`)
-a|
-`*` selector acts on arrays and objects.
-
-* Selecting all `user` values from an array:
-+
-```
-[ { "user" : "a", "user" : "b" } ].*user
-```
-+
-Returns `["a", "b"]` if the `output` directive is `application/json`.
-
-* Selecting all `user` values from an object:
-+
-```
-{ user : "a", "user" : "b" }.*user
-```
-+
-Returns: `["a", "b"]` if the `output` directive is `application/json`.
-
-* Selecting all `user` values from the `users` key:
-+
-```
-{ "users" : [ { "user" : "a" }, { "user" : "b" } ] }.users.*user`
-```
-+
-Returns: `[ "a", "b" ]` if the `output` directive is `application/json`.
-
-For more examples, see <<multi_value>>.
-
-| *Key-Value Pair* (`&amp;`)
-a|
-`&amp;` selector acts on arrays and objects.
-
-* Selecting values from the `&amp;` key:
-+
-```
-{ "users" : [ "user" : "ale", "something": "else", "user" : "joe" ] }.users.&user
-```
-+
-Returns `[ { "user": "ale" }, { "user": "joe" } ]` if the `output` directive
-is `application/json`.
-
-For more examples, see <<key_value>>.
-
-| *Descendants* (`..`)
-a|
-`..` selector acts on arrays and objects.
-
-* Selecting values from the `c` key:
-+
-```
-{ "user" : [{ "a" : "b"}, {"c" : [ {"aa" : "bb"}, { "cc": "dd" } ] } ] , "user" : "d" }..c
-```
-+
-Returns `[ [ { "aa": "bb" }, { "cc": "dd" } ] ]`.
-
-For more examples, see <<descendants>>.
-
-| *Index* (`[]`)
-a|
-`[]` selector acts on indices of an array.
+.DataWeave Script:
+[source,dataweave,linenums]
+----
+%dw 2.0
+var myObject = { "myKey" : "1234", "name" : "somebody" }
+var myArray = [ { "myKey" : "1234" }, { "name" : "somebody" } ]
+output application/json
+---
+{
+    selectingValueUsingKeyInObject : myObject.name,
+    selectingValueUsingKeyOfObjectInArray : myArray.name,
+}
+----
 
-* Selecting an index from an array:
-** `[1,2,3,4][0]` returns `[1]`
-+
-The index of the first element in an array is always `0`.
-+
-** `[1,2,3,4][4]` returns `[3]`
+.Output:
+[source,JSON,linenums]
+----
+{
+    "selectingValueUsingKeyInObject": "somebody",
+    "selectingValueUsingKeyOfObjectInArray": [ "somebody" ]
+ }
+----
 
-For more examples, see <<range>>.
+As the Output shows:
 
-| *Range* `[index1 to index2]`
-a| `[index1 to index2]` selector produces and array with values from selected
-indexes.
+* `myObject.name` returns the value `"somebody"`
+* `myArray.name` returns the array `[ "somebody" ]`
 
-* Selecting a range of indices:
-** `[0,1,2][0 to 1]` returns `[ 0, 1 ]`
-** `[0,1,2][-1 to 0]` returns an inverted list: `[ 2, 1, 0 ]`
+== Selector Quick Reference
 
-For more examples, see xref:dataweave-selectors#range-selector.adoc[Range Selectors]
+Single Value (`.`)::
+Acts on arrays and objects to return the value of a matching key. The syntax
+is `.myKey`. To retrieve values of duplicate keys in a DataWeave _object_, use
```

Comment:
Why is "object" in italics"?

Reply:
for emphasis, as opposed to array, on which selector behaves differently